src/Pages/OfficerDB/OfficerDB.js

```diff
@@ -1,75 +1,111 @@
-  row = (person, index) => {
-    return (
-      <tr key={index}>
-        <td>{parseInt(index) + 1}</td>
-        <td>{person.firstName}</td>
-        <td>{person.lastName}</td>
-        <td>{person.linkedIn}</td>
-        <td>{person.Github}</td>
-        <td>{person.description}</td>
-        <td>{person.major}</td>
-      </tr>
+  async componentDidMount () {
+    await this.getOfficers()
+    this.setState({ users: await api.getUsers(this.props.user.token) })
+  }
+
+  async getOfficers () {
+    this.setState(
+      { officers: await api.getOfficers(this.props.user.token) },
+      () => this.queryOfficers('')
     )
   }
 
+  async deleteOfficer (officerEmail) {
+    // delete officer!
+    await api.deleteOfficer(officerEmail, this.props.user.token)
+    // change user accessLevel to member-level
+    await api.editAccessLevel(officerEmail, 0, this.props.user.token)
+  }
+
+  setToggle () {
+    this.setState({ toggleAdder: !this.state.toggleAdder })
+  }
+
+  queryOfficers (query) {
+    query
+      ? this.setState({
+        queryOfficers: this.state.officers.filter(officer =>
+          officer.name
+            .trim()
+            .toLowerCase()
+            .includes(query)
+        )
+      })
+      : this.setState({ queryOfficers: this.state.officers })
+  }
+
   render () {
     return (
       <div className='App'>
+        <Input
+          style={{ marginBottom: '10px' }}
+          onChange={event =>
+            this.queryOfficers(event.target.value.trim().toLowerCase())}
+          placeholder='Search Name'
+        />
+
         <header className='App-header'>
           <h1>Officer Page</h1>
         </header>
         <Table striped bordered hover variant='dark'>
           <thead>
             <tr>
-              <th>#</th>
-              <th>First Name</th>
-              <th>Last Name</th>
-              <th>LinkedIn</th>
-              <th>Github</th>
-              <th>Description</th>
-              <th>Major</th>
+              {['Name', 'Role', 'LinkedIn', 'Github', 'Quote', 'Edit'].map(
+                (field, ind) => (
+                  <th key={ind}>{field}</th>
+                )
+              )}
+              <th
+                onClick={() => {
+                  this.setToggle()
+                }}
+              >
+                {svg.addSymbol()}
+              </th>
+              <AdderModal
+                user={this.props.user}
+                users={this.state.users}
+                toggle={this.state.toggleAdder}
+                getOfficers={() => this.getOfficers()}
+                setToggle={() => {
+                  this.setToggle()
+                }}
+              />
             </tr>
           </thead>
+
           <tbody>
-            {this.state.people.map((person, index) => {
-              return this.row(person, index)
-            })}
+            {this.state.queryOfficers.map((officer, index) => (
+              <TableRow
+                officer={officer}
```

Comment:
I am noticing that inside `TableRow` you are doing a lot of `props.officer.name`, `props.officer.role` etc.

To send the props over without being inside a `officer` object you can do

```suggestion
                {...officer}
```
you should then be able to access everything like `props.title`, `props.role`

Reply:
Because i will need to pass on the entire ```officer``` object to another component, so it is simpler this way to use it directly from ```props.officer``` rather than smaller ```{...officer}```

Reply:
hey man do this^^^^^^^^^^^^